health-checker/src/main/java/io/stargate/health/StorageHealthChecker.java

```diff
@@ -0,0 +1,142 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.health;
+
+import com.codahale.metrics.health.HealthCheck;
+import com.datastax.oss.driver.api.core.uuid.Uuids;
+import io.stargate.db.datastore.DataStore;
+import io.stargate.db.datastore.DataStoreFactory;
+import io.stargate.db.datastore.ResultSet;
+import io.stargate.db.datastore.Row;
+import io.stargate.db.query.Predicate;
+import io.stargate.db.query.builder.Replication;
+import io.stargate.db.schema.Column;
+import io.stargate.db.schema.ImmutableColumn;
+import io.stargate.db.schema.ImmutableTable;
+import io.stargate.db.schema.Table;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class StorageHealthChecker extends HealthCheck {
+  private static final Logger logger = LoggerFactory.getLogger(DataStoreHealthChecker.class);
+  private static final boolean SHOULD_CREATE_KS_AND_TABLE =
+      Boolean.parseBoolean(
+          System.getProperty("stargate.health_check.data_store.create_ks_and_table", "true"));
+  private static final String KEYSPACE_NAME =
+      System.getProperty("stargate.health_check.keyspace_name", "data_store_health_check");
+  private static final String TABLE_NAME =
+      System.getProperty("stargate.health_check.table_name", "health_table");
+  private static final int INSERT_TTL_SECONDS = 600;
+
+  static final String PK_COLUMN_NAME = "pk";
+  static final String VALUE_COLUMN_NAME = "value";
+  static final Table EXPECTED_TABLE =
+      ImmutableTable.builder()
+          .keyspace(KEYSPACE_NAME)
+          .name(TABLE_NAME)
+          .addColumns(
+              ImmutableColumn.create(
+                  PK_COLUMN_NAME, Column.Kind.PartitionKey, Column.Type.Timeuuid),
+              ImmutableColumn.create(VALUE_COLUMN_NAME, Column.Kind.Regular, Column.Type.Varchar))
+          .build();
+  private final DataStoreFactory dataStoreFactory;
+
+  public StorageHealthChecker(DataStoreFactory dataStoreFactory)
+      throws ExecutionException, InterruptedException {
+
+    this.dataStoreFactory = dataStoreFactory;
+    if (SHOULD_CREATE_KS_AND_TABLE) {
+      ensureTableExists(dataStoreFactory.createInternal());
+    }
+  }
+
+  @Override
+  protected Result check() throws Exception {
+    try {
+      DataStore dataStore = dataStoreFactory.createInternal();
+
+      UUID pkValue = Uuids.timeBased();
```

Comment:
I might be nit-picking, but I see a potential for unbounded data growth here if the health check is executed too often (some whatever reason). I know my suggestion to increase the TTL makes it worse :facepalm: 

What do you think about using logic like this:
* Have a static UUID PK value per Stargate node (changes on restart)
* Use current timestamp at the "test" value
* Insert with a user-provided write timestamp
* Read the value back
* If the value is what we write - healthy
* If the value is different - it might be a concurrent health check operation from the same node - check the the row timestamp, if it's greater that ours - healthy

WDYT?

Reply:
```
I might be nit-picking, but I see a potential for unbounded data growth here if the health check is executed too often 
```
Could you elaborate on how it may grow? Every health-check is inserting a new row with TTL = 10 minutes (after change) 
Assuming we will execute 1 health-check per second, this is `600` row maximum,. After this time they will be removed. 
Assuming that we have 10 Stargate nodes it will be `600 * 10`. So where's the problem?

Reply:
It might be a problem if the health check URL is hit with a high rate... this would be an abnormal use case, but I am worries about us opening backdoors for malicious use cases.

Reply:
```
If the value is different - it might be a concurrent health check operation from the same node - check the row timestamp, if it's greater that ours - healthy
```
I think that this is not possible, because the WHERE clause fetches exactly the same record that was inserted.
We may consider using the same CL for both write and read queries, maybe `LOCAL_QUORUM`?

Reply:
> It might be a problem if the health check URL is hit with a high rate... this would be an abnormal use case, but I am worried about us opening backdoors for malicious use cases.

If that the case, wouldn't it be a problem for all health-checks that we have? If this is a real-risk we should consider implementing a mechanism top of all health-checks to solve that problem. Not a work-around for it in this specific health-check (because other health-checks will have the same problem). wdyt?

Reply:
Other health checkers do not generate data, so they do not have this problem, I think. That's why I'm proposing to use a fixed PK value. This way the amount of data does not grow with high usage.

Reply:
By the way, this is the only remaining concern from my POV. Otherwise, the PR LGMT :+1: 

@dougwettlaufer : what's your take?

Reply:
+1 for fixed PK per stargate deployment. The default check for k8s is every 10 seconds so if you're running multiple stargate pods (as recommended) with each creating a new record every 10 seconds then I could definitely see data blowing up.

Reply:
@dougwettlaufer there is a write TTL. The problem that we discussing is with a malicious user that can call this endpoint N times per second. For your example of query per 10 second and ttl = 10 minutes even if we assume 20 nodes it will be `120 000` records. If we change the TTL to a lower value (i.e. 3 minutes) it's `36 000`. I don't such an amount of data is blowing up. I'll rework it.